device/esp_tinyusb/descriptors_control.c

```diff
@@ -89,11 +160,94 @@ uint16_t const *tud_descriptor_string_cb(uint8_t index, uint16_t langid)
 // =============================================================================
 // Driver functions
 // =============================================================================
-
-void tinyusb_set_descriptor(const tusb_desc_device_t *dev_desc, const char **str_desc, int str_desc_count, const uint8_t *cfg_desc)
+esp_err_t tinyusb_set_descriptors(const tinyusb_config_t *config)
 {
-    assert(dev_desc && str_desc && cfg_desc);
-    assert(str_desc_count <= USB_STRING_DESCRIPTOR_ARRAY_SIZE);
+    esp_err_t ret = ESP_FAIL;
+    assert(config);
+    const char **pstr_desc;
+    // Parse configuration and save descriptors's pointer
+    // Select Device Descriptor
+    if (NULL == config->device_descriptor) {
+        ESP_LOGW(TAG, "The device's device descriptor is not provided by user, using default.");
+    }
+    s_desc_cfg.dev = config->device_descriptor
+                     ? config->device_descriptor
+                     : &descriptor_dev_kconfig;
```

Comment:
Nitpick: IMO this is cleaner. Avoids the double conditional statements.

```suggestion
        s_desc_cfg.dev = &descriptor_dev_kconfig;
    } else {
        s_desc_cfg.dev = config->device_descriptor;
    }
```